modes/documentation-mode/internal.lisp

```diff
@@ -171,16 +175,18 @@
                 (insert-string point "-|")
                 (insert-character point #\newline)))))
 
-(defun generate-markdown-file (filename)
+(defgeneric generate-markdown-file (filename type))
```

Comment:
The method dispatch is very attractive and I love it.
But here I felt it would be better to simply separate the functions
What do you think?

Reply:
Ummm, I was thinking to use this functionality to generate other kind of markdown, like with maybe benchmark, so I think it can be reused.

But I tend to overuse generics, so if you like I can change it back to a function :+1:

Reply:
I was only thinking about statically determining which method the caller is.
I thought a function would be better in that case.
However, if we need extensibility in the future, it may be better to make it a method.

Reply:
>  However, if we need extensibility in the future, it may be better to make it a method.

Ummm, I do think it may be extensible in the future indeed, in any case I'll change it if you want :+1:

Reply:
Well, this might be fine as it is too.